sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -133,7 +139,7 @@ def upload_block_blob(  # pylint: disable=too-many-locals, too-many-statements
             total_size = length
             encryptor, padder = None, None
             if encryption_options and encryption_options.get('key'):
-                cek, iv, encryption_data = generate_blob_encryption_data(
+                cek, iv, encryption_data = generate_blob_encryption_data(  # type: ignore [assignment]
```

Comment:
What's this ignore for?

Reply:
`azure\storage\blob\_upload_helpers.py:142: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]`

[Here](https://github.com/vincenttran-msft/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_encryption.py#L801-L832) is a link to the relevant API if you think that that means `generate_blob_encryption_data` is mistyping encryption_data as `Optional[str]` (It really shouldn't be optional to my eyes, but we are blocked by `key_encryption_key` making it conditionally set. That can be resolved if we just knew a valid return value (or exception to raise) in the case that `key_encryption_key` is `None`.)

Reply:
I think the issue is that `encryption_data` is used above on line 108 where it gets the type of just `str` and now you are trying to assign it an `Optional[str]`. So, you should be able to solve that somehow... maybe assign a type on line 108, declare the variable higher up as None and assign it a type, or use another variable.

Reply:
You're so right 😄